app/src/main/java/com/example/henzoshimada/feeltrip/FeelTripApplication.java

```diff
@@ -59,4 +65,79 @@ public static Participant getParticipant(){
         return participant;
     }
 
+    private static FilterController filterController = null;
+    public static FilterController getFilterController(){
+        if (filterController == null){
+            filterController = new FilterController();
+        }
+        return filterController;
+    }
+
+    private static  MoodAdapter moodAdapter = null;
+    private static ArrayList<Mood> moodArrayList = new ArrayList<Mood>();
+    public static MoodAdapter getMoodAdapter(Context context) {
+        if (moodAdapter == null) {
+            moodAdapter = new MoodAdapter(moodArrayList, context.getApplicationContext());
+        }
+        return moodAdapter;
+    }
+
+    public static ArrayList<Mood> getMoodArrayList() {
+        if(moodArrayList == null) {
+            moodArrayList = new ArrayList<Mood>();
+        }
+        return moodArrayList;
+    }
+
+    public static void setMoodArrayList(ArrayList<Mood> moodArrayList) {
+        FeelTripApplication.moodArrayList = moodArrayList;
+    }
+
+    private static int frag; // 0 for  Home frag, 1 for Profile frag, 2 for Map frag
+    public static void setFrag(String fragstr) {
+        switch(fragstr) {
+            case "main":
+                frag = 0;
+                break;
+            case "profile":
+                frag = 1;
+                break;
+            case "map":
+                frag = 2;
+                break;
+            default:
+                Log.i("Error", "The given search mode is invalid.");
+                return;
+        }
+    }
+
+    public static void loadFromElasticSearch(){
+        Log.d("listTag", "load from ES");
+        moodArrayList.clear();
+        ElasticSearchController.GetFilteredMoodsTask getMoodTask;
```

Comment:
For the purposes of updating the data on the MoodAdapter no matter where we are in the app.